frontend/components/forms/image-uploader/types.ts

```diff
@@ -0,0 +1,18 @@
+import { FieldPath, Path, RegisterOptions, UseFormRegister } from 'react-hook-form';
+
+export type ImageUploaderProps<FormValues> = {
+  /** name of the input */
+  name: Path<FormValues>;
+  /** id of the input */
+  id: string;
+  /** React Hook Form's `register` function */
+  register: UseFormRegister<FormValues>;
+  /** Options for React Hook Form's `register` function */
+  registerOptions?: RegisterOptions<FormValues, FieldPath<FormValues>>;
```

Comment:
🟠

If you overwrite `registerOptions.onChange` in the component, you should remove the possibility to pass it externally:
```suggestion
  registerOptions?: Omit<RegisterOptions<FormValues, FieldPath<FormValues>>, 'onChange'>;
```